src/state/ledger/diff/account.rs

```diff
@@ -162,3 +162,132 @@ impl std::fmt::Debug for UpdateType {
+        let source_public_key = source_public_key_result.clone();
+
+        let receiver_str = "B62qjoDXHMPZx8AACUrdaKVyDcn7uxbym1kxodgMXztn6iJC2yqEKbs";
+        let receiver_public_key_result = PublicKey::from_address(receiver_str).unwrap();
+        let receiver_public_key = receiver_public_key_result.clone();
+
+        let payment_command = Command::Payment(Payment {
+            source: source_public_key.into(),
+            receiver: receiver_public_key.into(),
+            amount: 536900000000,
+        });
+
+        let expected_result = vec![
+            AccountDiff::Payment(PaymentDiff {
+                public_key: source_public_key_result.into(),
+                amount: 536900000000,
+                update_type: UpdateType::Deduction,
+            }),
+            AccountDiff::Payment(PaymentDiff {
+                public_key: receiver_public_key_result.into(),
+                amount: 536900000000,
+                update_type: UpdateType::Deposit,
+            }),
+        ];
+
+        assert_eq!(AccountDiff::from_command(payment_command), expected_result);
+    }
+
+    #[test]
+    fn test_from_command_delegation() {
+        let delegator_str = "B62qpYZ5BUaXq7gkUksirDA5c7okVMBY6VrQbj7YHLARWiBvu6A2fqi";
+        let delegator_public_key_result = PublicKey::from_address(delegator_str).unwrap();
+        let delegator_public_key = delegator_public_key_result.clone();
+
+        let delegate_str = "B62qjSytpSK7aEauBprjXDSZwc9ai4YMv9tpmXLQK14Vy941YV36rMz";
+        let delegate_public_key_result = PublicKey::from_address(delegate_str).unwrap();
+        let delegate_public_key = delegate_public_key_result.clone();
+
+        let delegation_command = Command::Delegation(Delegation {
+            delegator: delegator_public_key.into(),
+            delegate: delegate_public_key.into(),
+        });
+
+        let expected_result = vec![AccountDiff::Delegation(DelegationDiff {
+            delegator: delegator_public_key_result.into(),
+            delegate: delegate_public_key_result.into(),
+        })];
+
+        assert_eq!(
+            AccountDiff::from_command(delegation_command),
+            expected_result
+        );
+    }
+
+    #[test]
+    fn test_from_coinbase() {
+        let coinbase_receiver_str = "B62qospDjUj43x2yMKiNehojWWRUsE1wpdUDVpfxH8V3n5Y1QgJKFfw";
+        let coinbase_receiver_result = PublicKey::from_address(coinbase_receiver_str).unwrap();
+        let coinbase_receiver = coinbase_receiver_result.clone();
+
+        let supercharge_coinbase = true;
+
+        let account_diff =
+            AccountDiff::from_coinbase(coinbase_receiver_result.into(), supercharge_coinbase);
+
+        let expected_payment_diff = PaymentDiff {
+            public_key: coinbase_receiver.into(),
+            amount: 1440 * (1e9 as u64),
+            update_type: UpdateType::Deposit,
+        };
+        let expected_account_diff = AccountDiff::Payment(expected_payment_diff);
```

Comment:
Interesting. This could explain why our nonces were off.

Reply:
We're processing coinbases as payments? Payments should increase the nonce, but I don't think coinbases should. Not sure. Just a thought.